src/plugins/event-plugins/__tests__/FetchPlugin.test.ts

```diff
@@ -969,4 +966,58 @@ describe('FetchPlugin tests', () => {
             segment_id: existingSegmentId
         });
     });
+
+    test('when urlsToTrace is empty then the plugin does not record a trace', async () => {
+        // Init
+        const config: PartialHttpPluginConfig = {
+            urlsToTrace: []
+        };
+
+        const plugin: FetchPlugin = new FetchPlugin(config);
+        plugin.load(xRayOnContext);
+
+        // Run
+        await fetch(URL);
+        plugin.disable();
+
+        // Assert
+        expect(record).toHaveBeenCalledTimes(0);
+    });
+
+    test('when the url does not match urlsToTrace then the plugin does not record a trace', async () => {
+        // Init
+        const config: PartialHttpPluginConfig = {
+            recordAllRequests: true,
+            urlsToTrace: [/a^/]
+        };
+
+        const plugin: FetchPlugin = new FetchPlugin(config);
+        plugin.load(xRayOnContext);
+
+        // Run
+        await fetch(URL);
+        plugin.disable();
+
+        // Assert
+        expect(record).toHaveBeenCalledTimes(1);
+        expect(record.mock.calls[0][0]).toEqual(HTTP_EVENT_TYPE);
+    });
+
+    test('when the url matches urlsToTrace then the plugin records a trace', async () => {
+        // Init
+        const config: PartialHttpPluginConfig = {
+            urlsToTrace: [/https:\/\/aws\.amazon\.com/]
+        };
+
+        const plugin: FetchPlugin = new FetchPlugin(config);
+        plugin.load(xRayOnContext);
+
+        // Run
+        await fetch(URL);
+        plugin.disable();
+
+        // Assert
+        expect(record).toHaveBeenCalledTimes(1);
+        expect(record.mock.calls[0][0]).toEqual(XRAY_TRACE_EVENT_TYPE);
```

Comment:
question: what about test case for if the regex expression is invalid?

Reply:
In general the web client doesn't validate config options, which means the behavior of the web client is undefined when a config option is invalid.

Perhaps we should consider doing validation for options on critical paths. The downside is that it makes the package larger and more complex.

Reply:
I think an invalid reg exp would throw an error here at the very least https://github.com/aws-observability/aws-rum-web/blob/main/src/plugins/utils/http-utils.ts#L78